lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -410,16 +412,21 @@ public function processParameterValue($value)
             return $value;
         }
 
-        if (is_object($value) && $this->_em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
+        if ($value instanceof Mapping\ClassMetadata) {
+            return $value->name;
+        }
+
+        if ( ! is_object($value)) {
+            return $value;
+        }
+
+        try {
             $value = $this->_em->getUnitOfWork()->getSingleIdentifierValue($value);
 
             if ($value === null) {
                 throw ORMInvalidArgumentException::invalidIdentifierBindingEntity();
             }
-        }
-
-        if ($value instanceof Mapping\ClassMetadata) {
-            return $value->name;
+        } catch (MappingException|ORMMappingException $e) {
```

Comment:
Can you please add a comment explaining why these exceptions are ignored?

Reply:
```suggestion
        } catch (MappingException | ORMMappingException $e) {
```

Reply:
@Majkl578 we currently don't have a rule for that in Doctrine CS, so I'd like to have that discussion in `doctrine/coding-standard` before committing to it.

Reply:
@Majkl578 Do you know any Sniff for that? I agree with @alcaeus, we should discuss it on the repo.